src/test/scala/chiselTests/ChiselSpec.scala

```diff
@@ -25,18 +27,88 @@ import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+    val workspacePath = Seq(
+      "test_run_dir",
+      getClass.getSimpleName,
+      /// This is taken from the legacy `TesterDriver` class. It isn't ideal and we hope to improve this eventually.
+      timeStampFormat.format(Calendar.getInstance().getTime())
+    ).mkString("/")
+    val workspace = new Workspace(workspacePath)
+    workspace.reset()
+    val elaboratedModule = workspace.elaborateGeneratedModule({ () => t })
+    additionalVResources.foreach(workspace.addPrimarySourceFromResource(getClass(), _))
+    workspace.generateAdditionalSources()
+    val simulation = workspace.compile(verilatorBackend)(
+      "verilator", {
+        import CommonCompilationSettings._
+        CommonCompilationSettings(
+          availableParallelism = AvailableParallelism.UpTo(Runtime.getRuntime().availableProcessors()),
+          optimizationStyle = OptimizationStyle.OptimizeForCompilationSpeed,
+          verilogPreprocessorDefines = Seq(
+            VerilogPreprocessorDefine("ASSERT_VERBOSE_COND", s"!${Workspace.testbenchModuleName}.reset"),
+            VerilogPreprocessorDefine("PRINTF_COND", s"!${Workspace.testbenchModuleName}.reset"),
+            VerilogPreprocessorDefine("STOP_COND", s"!${Workspace.testbenchModuleName}.reset")
+          )
+        )
+      },
+      // Seq("--assert", "-Wno-fatal", "-Wno-WIDTH", "-Wno-STMTDLY")
+      verilator.Backend
+        .CompilationSettings(disabledWarnings = Seq("WIDTH", "STMTDLY"), disableFatalExitOnWarnings = true),
+      customSimulationWorkingDirectory = None,
+      verbose = false
+    )
+    try {
+      simulation
+        .runElaboratedModule(elaboratedModule) { module =>
+          val dut = module.wrapped
+          val clock = module.port(dut.clock)
+          val reset = module.port(dut.reset)
+          reset.set(1)
+          clock.tick(
+            timestepsPerPhase = 1,
+            maxCycles = 10,
+            inPhaseValue = 1,
+            outOfPhaseValue = 0,
+            sentinel = None
+          )
+          reset.set(0)
+
+          val elapsedCycles = clock.tick(
+            timestepsPerPhase = 1,
+            maxCycles = 10000,
+            inPhaseValue = 1,
+            outOfPhaseValue = 0,
+            sentinel = None
+          )
+        }
+      true
+    } catch {
+      // We eventually want to have a more structured way of detecting assertions, but this works for now.
+      case svsim.Simulation.UnexpectedEndOfMessages =>
+        val filename = s"$workspacePath/workdir-verilator/simulation-log.txt"
+        for (line <- scala.io.Source.fromFile(filename).getLines()) {
+          if (line.contains("Verilog $finish")) {
+            // We don't immediately exit on $finish, so we need to ignore assertions that happen after a call to $finish
+            return true
+          }
+          if (line.contains("Assertion failed")) {
+            return false
+          }
+        }
+        return true
```

Comment:
This isn't ideal, and we eventually want to have more structured detection of assertions and `$finish`. I leave it up to the maintainers whether or not we commit this hack now or wait for better finish/assertion support.